lib/server/services/qualityControlFlag/QcFlagTypeService.js

```diff
@@ -95,6 +95,55 @@ class QcFlagTypeService {
         return qcFlagTypeAdapter.toEntity(await this.getById(newFlagInstance.id));
     }
 
+    /**
+     * Update existing QC Flag Type
+     * @param {number} [id] QC flag type id
+     * @param {Partial<QcFlagType>} [patch] QC flag type parameters
+     * @param {object} [relations] QC flag type entity relations
+     * @param {Partial<UserIdentifier>} [relations.user] user identifiers
+     * @return {Promise<QCFlagType>} promise
+     */
+    async update(id, patch, relations = {}) {
+        const {
+            name,
+            method,
+            bad,
+            color,
+            archived,
+        } = patch;
+        const { user: { userId, externalUserId } = {} } = relations;
+
+        const user = await getUserOrFail({ userId, externalUserId });
+
+        const qcFlagType = await QcFlagTypeRepository.findOne({ where: { id } });
+        if (!qcFlagType) {
+            throw new NotFoundError(`Quality Control Flag Type with this id (${id}) could not be found`);
+        }
+
+        const existingQcFlagType = await QcFlagTypeRepository.findOne({
+            where: { [Op.not]: { id }, [Op.or]: [name ? { name } : {}, method ? { method } : {}] },
+        });
+        if (existingQcFlagType) {
+            const failureReason = [name ? `name ${name}` : null, method ? `method ${method}` : null].filter((_) => _).join(' or ');
+            throw new ConflictError(`A QC flag with ${failureReason} already exists`);
+        }
+
+        await QcFlagTypeRepository.update(qcFlagType, {
+            name,
+            method,
+            bad,
+            color,
+            lastUpdatedById: user.id,
+            archivedAt: archived === true && !qcFlagType.archvied
```

Comment:
Have you tested this? There is a typo, you should have noticed

Reply:
fixed, test extended